### PR TITLE
[client][some api] build interface for the status step

### PR DESCRIPTION
**Pull request by WilhelmWillie**
Screen Shot 2019-10-30 at 2 12 06 PM

The status step is the first step in the application process. This is the default screen that shows where the user is in the application process.

### CODE REVIEW

File: lib/authenticate.tsx [hunk 1]
```diff
@@ -16,6 +17,21 @@ export async function getUser(req) {
   return req ? req.user : null;
 }
 
+export async function getProfile(req) {
+  const rawProfileData = await fetch("http://localhost:3000/api/profile", {
+    headers: {
+      Cookie: req.headers.cookie
```
Comment (WilhelmWillie) [inline lib/authenticate.tsx]: need to include cookie from request for SSR. otherwise, `/api/profile` will 403

File: api/models/hackerProfile.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// If you make any changes to HackerProfile, make sure you do the following:
+// 1) Generate a Sequelize migration that adds/removes columns as needed
+// 2) Update the Profile type definition in odyssey.d.ts
```
Comment (WilhelmWillie) [inline api/models/hackerProfile.js]: in an ideal world w/ time, i'd write some lint thing for this but i'm not that much of a nerd

File: api/hackerProfile.js [hunk 3]
```diff
@@ -5,11 +5,17 @@ const router = express.Router();
 const Sentry = require("@sentry/node");
 
 router.get("/", authMiddleware, async (req, res) => {
-  const hackerProfile = await models.HackerProfile.findAll({
+  console.log("AM I EVEN CALLED");
```
Comment (WilhelmWillie) [inline api/hackerProfile.js]: lmao im a mess

File: lib/authenticate.tsx (hunk 1 above)
Comment (saresend) [reply lib/authenticate.tsx]:
```javascript
const rawProfileData = await fetch("http://localhost:3000/api/profile", {
    headers: req.headers,
});
```
Not sure what else is going down in the headers, but we could just do this? For some reason I thought the headers were already populated

Comment (WilhelmWillie) [reply lib/authenticate.tsx]: they do for client-side routing but this is for server-side rendering where the server itself is doing the `fetch` call, which is why we need to grab the `req.headers` from the client `req` 

good call tho, will do that instead

Comment (saresend) [reply lib/authenticate.tsx]: Ah gotcha, kk

File: api/hackerProfile.js (hunk 3 above)
Comment (saresend) [reply api/hackerProfile.js]: r u ok

**Pull request merged**